staging/src/k8s.io/apiserver/pkg/admission/plugin/webhook/admission.go

```diff
@@ -187,6 +192,32 @@ func (a *GenericAdmissionWebhook) loadConfiguration(attr admission.Attributes) (
 	return hookConfig, nil
 }
 
+type versioned struct {
+	admission.Attributes
+	oldObject runtime.Object
+	object    runtime.Object
+}
+
+func (v versioned) GetObject() runtime.Object {
+	return v.object
+}
+
+func (v versioned) GetOldObject() runtime.Object {
+	return v.oldObject
+}
+
+func (a *GenericAdmissionWebhook) convertToGVK(obj runtime.Object, gvk schema.GroupVersionKind) (runtime.Object, error) {
```

Comment:
I'd like to see some unit tests exercising this code that include legacy kinds, kinds across groups, and unstructured/CRD kinds.

@sttts I think this looks pretty good, but I'm concerned about CRDs in particular.  How does it look to you?

Reply:
Sure. The passed in CRD object is of `Unstructured` type, with GVK equal to the `attr.GetKind()`, so this function will be a no-op. I'll add a unit test to confirm.

I think we eventually need an e2e test for webhooks on CRD.

Reply:
Added unit tests as well.